### ISSUE TITLE
KubernetesDeployer may report incorrect target namespace

### ISSUE THREAD

**Issue by tdiesler** [quarkusio/quarkus#43527]
### Describe the bug

I'm running ...

```
mvnw --file .camel-jbang-run/timer-log -Dquarkus.container-image.build=true -Dquarkus.container-image.push=false -Dquarkus.kubernetes.deploy=true -Dquarkus.kubernetes.namespace=foo package
```

which correctly builds and deploys the container to k8s namespace foo.

From the build log I seems that the container is deployed to namespace default

```
[INFO] [io.quarkus.container.image.jib.deployment.JibProcessor] Starting (local) container image build for jar using jib.
[INFO] [io.quarkus.container.image.jib.deployment.JibProcessor] Using docker to run the native image builder
[WARNING] [io.quarkus.container.image.jib.deployment.JibProcessor] Base image 'registry.access.redhat.com/ubi8/openjdk-17-runtime:1.20' does not use a specific image digest - build may not be reproducible
[INFO] [io.quarkus.container.image.jib.deployment.JibProcessor] Using base image with digest: sha256:e2f33a6c60db4f4e70882a4a557eec5890997f8a1be7e3eb8971a0ff8a45a1a8
[INFO] [io.quarkus.container.image.jib.deployment.JibProcessor] Container entrypoint set to [/opt/jboss/container/java/run/run-java.sh]
[INFO] [io.quarkus.container.image.jib.deployment.JibProcessor] Created container image examples/timer-log:1.0.0 (sha256:fdfbd1315086e3032c41e87d6488a541a59ae970cf81f65e2675973c174ec171)

[INFO] [io.quarkus.kubernetes.deployment.KubernetesDeployer] Deploying to kubernetes server: https://kubernetes.docker.internal:6443/ in namespace: default.
[INFO] [io.quarkus.kubernetes.deployment.KubernetesDeployer] Applied: Service timer-log.
[INFO] [io.quarkus.kubernetes.deployment.KubernetesDeployer] Applied: Deployment timer-log.
[INFO] [io.quarkus.deployment.QuarkusAugmentor] Quarkus augmentation completed in 4661ms
```

**Comment by geoand** [comment]
@metacosm is this something you fixed recently?

**Comment by metacosm** [comment]
> @metacosm is this something you fixed recently?

No, the last thing I worked on somewhat related was https://github.com/quarkusio/quarkus/issues/29809 (at least, that I can recall).

**Comment by tdiesler** [comment]
Also, please confirm that there is a path that allows the user to generate a manifest **without** explicit namespace, which can later be deployed to any namespace using `-Dquarkus.kubernetes.namespace`

**Comment by iocanel** [comment]
So, the idea is that by default we add no namespace to the generated resources.
This allows them to be deployed to the `current` namespace (the namespace that is set in the current context of your kubectl).
AFAIR, this is the namespace that is reported by the command.

One can change that namespace either by using `kubectl` or `oc` to change namespace, or by setting `quarkus.kubernetes.namespace` which will result in adding the namespace in the generated resoruces.

Now, we could possibly change the messge if `quarkus.kubernetes.namespace` is set, but I am wondering if this could cause more confusion.

**Comment by metacosm** [comment]
> Now, we could possibly change the messge if `quarkus.kubernetes.namespace` is set, but I am wondering if this could cause more confusion.

Why would it cause more confusion? As I understand it the message currently uses the namespace to which `kubectl` is connected, correct? However, if you set `quarkus.kubernetes.namespace` in conjunction with `deploy` then Quarkus will override the default kube connection and deploy to the namespace pointed at by the `quarkus.kubernetes.namespace`, right?  Or am I misunderstanding things? If the above is correct, then the message should definitely be changed to say which namespace is actually targeted. Maybe even a warning could be issued to say that there's a discrepancy between the target namespace and the one the user is currently connected to (though that might be overdoing it, maybe).
